activemodel/lib/active_model/type/serialize_cast_value.rb

```diff
@@ -6,8 +6,12 @@ module SerializeCastValue # :nodoc:
       def self.included(klass)
         unless klass.respond_to?(:included_serialize_cast_value)
           klass.singleton_class.attr_accessor :included_serialize_cast_value
+          klass.silence_redefinition_of_method(:itself_if_class_included_serialize_cast_value)
           klass.attr_reader :itself_if_class_included_serialize_cast_value
         end
+        # unless klass.method_defined?(:itself_if_class_included_serialize_cast_value)
+        #   klass.attr_reader :itself_if_class_included_serialize_cast_value
+        # end
```

Comment:
Hum, I don't know if it's a bug, it really depends on how you look at it.

`DelegateClass` create a class with explicitly defined methods, so if you redefine them you get a warning. That being said, it's somewhat expected when you create a delegator that you'd redefine some of them, so yes, kind of a bug.

As for the fix, you have two solutions:

  - Define the methods in an ancestor instead (either superclass or included module)
  - Use the self alias trick `alias :foo, :foo` to silence the warnings.

Would be worth opening an issue upstream IMO: https://bugs.ruby-lang.org/projects/ruby-master/issues. If you don't feel like it, I can do it.

Reply:
> If I understand your suggestion correctly, that would not work in this case because the delegated methods in a DelegateClass class come before ancestor methods. 

Sorry, my comment was very confusing. What I was suggesting here was a "bug fix" PR to `DelegateClass` so that it defines these methods in an ancestor.